filebeat/harvester/reader/line.go

```diff
@@ -13,7 +13,7 @@ import (
 // using the configured codec. The reader keeps track of bytes consumed
 // from raw input stream for every decoded line.
 type Line struct {
-	rawInput   io.Reader
+	rawInput   io.ReadCloser
```

Comment:
+1 on the NopClosed option as so far almost all readers needed the close method. The main reason is that readers can be chained.